CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -0,0 +1,536 @@
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(default(List<long>)))
+                .ReturnsAsync(default(List<long>));
+            
+            dashbordrepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(
+               new List<long>() { 6, 7, 8, 9, 10 }, new List<long>() { 2 }))
+                .ReturnsAsync(new List<AverageStudentMarkDto>() {
+                    new AverageStudentMarkDto()
+                    {
+                        CourseId = 2,
+                        StudentGroupId = 2,
+                        StudentId = 6,
+                        StudentAverageMark = 5.1
+                    }
+                });
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(default(List<long>), default(List<long>)))
+                .ReturnsAsync(new List<AverageStudentMarkDto>());
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentsAverageVisitsByStudentIdsAndGroupsIdsAsync(
+                 new List<long>() { 6, 7, 8, 9, 10 }, new List<long>() { 2 }))
+                .ReturnsAsync(new List<AverageStudentVisitsDto>()
+                {
+                    new AverageStudentVisitsDto()
+                    {
+                        CourseId = 2,
+                        StudentGroupId = 2,
+                        StudentId = 6,
+                        StudentAverageVisitsPercentage = 15
+                    }
+                });
+
+            dashbordrepositoryMock.Setup(x => x.GetStudentsAverageVisitsByStudentIdsAndGroupsIdsAsync(default(List<long>), default(List<long>)))
+                .ReturnsAsync(new List<AverageStudentVisitsDto>());
+
+            _unitOfWorkMock.Setup(x => x.DashboardRepository).Returns(dashbordrepositoryMock.Object);
+
+            var dashbordservice = new DashboardService(_unitOfWorkMock.Object);
+
+            //Act
+            var resultWithData = await dashbordservice.GetStudentsResultAsync(studentResultRequestWithData);
+            var resultWithWrongData = await dashbordservice.GetStudentsResultAsync(studentResultWrongRequest);
+            var resultWithoutStudent = await dashbordservice.GetStudentsResultAsync(studentResultRequestWithOutStudent);
+
+            //Assert
+            Assert.NotNull(resultWithData.Data);
+            Assert.Equal(ErrorCode.ValidationError, resultWithWrongData.Error.Code);
+            Assert.Empty(resultWithoutStudent.Data.AverageStudentsMarks);
+            Assert.Empty(resultWithoutStudent.Data.AverageStudentVisits);
+        }
+
+        [Fact]
+        public async Task GetStudentClassbook()
+        {
+            //Arrange
+            long studentIdWithGroup = 7;
+            long studentIdWithoutGroup = 20;
+            
+            var claims = new List<Claim>()
+            {
+                new Claim(ClaimTypes.Name, "Denys"),
+                new Claim(ClaimTypes.NameIdentifier , "1"),
```

Comment:
Why do we need this claim?

Reply:
we need second claim for group without student because we can not use one claim for 2 methods its will make Error in Validation methods